iothub/device/src/Authentication/IClientIdentity.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Azure.Devices.Client.Transport;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// Interface for device configurations and common attributes.
+    /// This has been included for our own unit testing.
+    /// </summary>
+    internal interface IClientIdentity : IAuthorizationProvider
+    {
+        AuthenticationWithTokenRefresh TokenRefresher { get; }
+
+        string IotHubName { get; }
+
+        string DeviceId { get; }
+
+        string ModuleId { get; }
+
+        string HostName { get; }
+
+        Uri HttpsEndpoint { get; }
+
+        Uri AmqpEndpoint { get; }
```

Comment:
Won't we use the same endpoint for both Https and Amqp? I get that we'll have some http/amqp specific suffixes and/or prefixes, but I'd envision that being known within the amqp/http layer

Reply:
You're correct in that. This class still has a bunch of things that we expect will change as we refactor the InternalClient implementation. We'll make those changes as we go. I didn't change all of it in this PR so as to avoid merge conflicts with the changes that David is working on.
Having said that, feel free to highlight things that look off so that we can keep those in mind as we refactor the other classes around this implementation.

Reply:
Gotcha, I didn't realize this was still a temporary field. No objections here then